### DISCUSSION TITLE
Livewire #[Url] attribute

### DISCUSSION THREAD

**Discussion by mschuyler** [livewire/livewire#7818] [1. Help] [answered]
I am creating a datatable that tracks multiple properties with the #[Url] attribute.

```
    #[Url]
    public string $search = '';

    #[Url]
    public array $sorts = [];

    #[Url]
    public int $perPage = 15;
```

When I modify any of these properties they are updated in the URL as expected. The issue is when I send a link to another user. 

Let's use this URL for example: website.com/?search=123&sorts[created_at]=asc&perPage=25

When we hit enter it will set all (3) properties correctly but will remove everything except ?search=123 from the URL. Regardless of everything I have tried search always works and no other property will work with #[Url] attribute. 

Any ideas what could be the issue?

**Comment by calebporzio** [comment]
Are you setting them in mount? Because that with rerun on the new page and override URL changes

**Reply by mschuyler** [reply]
Hi Caleb- thanks for replying. Yes I'm setting it in the mount as below -

```
    public function mount()
    {
        $this->perPage = request()->query('perPage', 15);
        $this->sorts = request()->query('sorts', []);
    }
```

When I make ```$perPage``` a string rather than int it works as expected - sets the property as expected and keeps it in the URL. 

So in my experience the array and int properties will only set the data and be removed from the URL.

**Comment by mschuyler** [comment, marked as answer]
I haven't had time to look into why but adjusting the attributes as below fixes my issue for those that run into it.

```
    #[Url]
    public string $search = '';

    #[Url(except: '')]
    public array $sorts = [];

    #[Url(except: '')]
    public int $perPage = 15;
```

This also doesn't require setting the properties in mount with the data from the request.

**Reply by joshhanley** [reply]
@mschuyler What patch version of livewire are you on? We recently changed something similar this so curious if you have an older version or the new one?

**Reply by mschuyler** [reply]
Livewire 3.4.2
Laravel 10.42
PHP 8.2.14

**Reply by joshhanley** [reply]
@mschuyler Ok thanks it's probably something else wrong but I'll double check I didn't break anything 😆

**Reply by joshhanley** [reply]
@mschuyler so it looks like I may have broken it with PR #7779. Will check with Caleb to see if he wants to revert that change.

**Reply by joshhanley** [reply]
@mschuyler have submitted a failing test PR #7827 to see whether Caleb wants to revert my changes or if we try and fix both scenarios.
